plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -28,7 +28,25 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-NEVRA = namedtuple('NEVRA', ['name', 'epoch', 'version', 'release', 'arch'])
+_NEVRA = namedtuple('NEVRA', ['name', 'epoch', 'version', 'release', 'arch'])
+
+
+class NEVRA(_NEVRA):
+    @classmethod
+    def _fromdict(cls, nevra_dict):
+        """Given a dict with NEVRA keys, return a NEVRA namedtuple using its values
+
+        :param nevra_dict: dict with NEVRA keys (name, epoch, version, release, arch)
```

Comment:
I am not sure if it is worth mentioning or not but the `nevra_dict` param should be a dictionary with those 5 keys as a minimun, it can have other keys as well like in  [`_get_pkglists_to_publish`](https://github.com/pulp/pulp_rpm/pull/1029/files#diff-4b7e211764045e0506c6ad263fddf8e0R677) and it will filter them out.